release.sh

```diff
@@ -5,6 +5,38 @@ cd $DIR
 
 set -e
 
+git remote update
+
+upstream=${1:-'@{u}'}
+remote=$(git rev-parse "$upstream")
+local=$(git rev-parse @)
+
+# check if local commit syncs with remote 
+if [ "$remote" != "$local" ]
+    then
+        echo "Error: local commit does not match remote. Exiting release script." 
+        exit 1
+fi 
+
+remoteTags=$(mktemp)
+git ls-remote --tags origin | grep -v '{}' | sort | tr -d [[:blank:]] >> $remoteTags
+
+localTags=$(mktemp)
+git show-ref --tags | grep -v '{}' | grep -v 'origin' | sort | tr -d [[:blank:]] >> $localTags
+
+# check if local tags match remote tags
+match=$(diff "$remoteTags" "$localTags")
```

Comment:
This is great! This would check that every tag is on the exact same commit both locally and remotely. Thanks